src/core/utils.ts

```diff
@@ -0,0 +1,110 @@
+import base32 from 'hi-base32'
+import {
+  ALGORAND_ADDRESS_BYTE_LENGTH,
+  ALGORAND_ADDRESS_LENGTH,
+  ALGORAND_CHECKSUM_BYTE_LENGTH,
+  CHECKSUM_ADDRESS_ERROR_MSG,
+  MALFORMED_ADDRESS_ERROR_MSG,
+} from './constants'
+import * as nacl from './nacl/naclWrappers'
+
+/**
+ * Check whether the environment is ReactNative
+ * @returns True if ReactNative, false otherwise
+ */
+export function isReactNative() {
+  const { navigator } = globalThis as { navigator?: { product?: string } }
+  if (typeof navigator === 'object' && navigator.product === 'ReactNative') {
+    return true
+  }
+  return false
+}
+
+export function arrayEqual<T>(a: ArrayLike<T>, b: ArrayLike<T>): boolean {
+  if (a.length !== b.length) {
+    return false
+  }
+  return Array.from(a).every((val, i) => val === b[i])
+}
+
+export function concatArrays(...arrs: ArrayLike<number>[]) {
+  const size = arrs.reduce((sum, arr) => sum + arr.length, 0)
+  const c = new Uint8Array(size)
+
+  let offset = 0
+  for (let i = 0; i < arrs.length; i++) {
+    c.set(arrs[i], offset)
+    offset += arrs[i].length
+  }
+
+  return c
+}
+
+/**
+ * bigIntToBytes converts a BigInt to a big-endian Uint8Array for encoding.
+ * @param bi - The bigint to convert.
+ * @param size - The size of the resulting byte array.
+ * @returns A byte array containing the big-endian encoding of the input bigint
+ */
+export function bigIntToBytes(bi: bigint | number, size: number) {
```

Comment:
I think these bigint to bytes are potentially abi specific, so you could potentially move into the abi namespace into a bigint.ts file?